src/experiments.js

```diff
@@ -364,6 +365,21 @@ export function getExperimentBranch(win, experimentName) {
     : null;
 }
 
+/**
+ * Returns an object containing all active experiment branches on the
+ * top Window.
+ *
+ * @param {!Window} win Window context to check for experiment state.
+ * @return {!Object} contains all experiment branches and their ids.
+ */
+export function getActiveExperimentBranches(win) {
+  const topWin = getTopWindow(win);
+  if (!topWin.__AMP_EXPERIMENT_BRANCHES) {
+    topWin.__AMP_EXPERIMENT_BRANCHES = {};
+  }
+  return topWin.__AMP_EXPERIMENT_BRANCHES;
```

Comment:
Are you sure you want to return the object reference instead of an array of keys or a shallow copy to prevent unintended mutations?

Reply:
Switched to shallow copy.